### PR TITLE
[PS]DDP-8559: More fixes to medical release v2 language that depends on whether is a child or self 

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
More fixes to medical release v2
There are a couple of problems remaining with this activity that I need some help with.
One is that ticket requests we do the age checking to ensure AOM rules and workflow are not violated.
@Trimax I would welcome your assistance here. I expect the checks you have been adding to consent/assent would be the same here.
The other problem that I just noticed is that we have required questions in the first page; we need to make sure that either there is a picklist item from one question chosen (MRR_RECORD_SEND_OPTIONS) or the single picklist item (MRR_STAFF_GETS_RECORDS_DIRECTLY) from another question is chosen.
I am not sure how we have done this in the past. @ssettipalli . I am hoping you can provide some guidance on this item.

### CODE REVIEW

File: pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf [hunk 1]
```diff
@@ -1623,19 +1623,19 @@
...
+    "medical_record_release_our_staff_contact_directly": "If you are unable to provide medical records through any of the options noted above, you may choose to have the study staff reach out to $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child's\", \"your\") physician(s) directly. Please note that this method will result in significant delays in $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child's\", \"your\") participation in the study."
+    "contact_for_help": "$ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"Parents or legal guardians\", \"Participants\") with questions about this form may contact a member of the study staff at <a href='tel:650-561-6750' class='Link'>650-561-6750</a>, Monday - Friday 9 a.m. - 5 p.m. or email at <a href='mailto:contact@projectsingular.org' class='Link'>contact@projectsingular.org</a>";
     "physician_list_intro": "Enter the physician, hospital, clinic, or provider where you would like $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child's\", \"your\") medical records to be sent from.",
```
Comment (MocanaAtBroad) [inline pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf:1628]: Mostly fixes where language varies depending on this being self or child. These are typical.

File: pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf [hunk 2]
```diff
@@ -1623,19 +1623,19 @@
...
+      "titleText": "By completing this information, you are agreeing to allow us to receive $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child’s\", \"your\") medical records through the option you selected above (sending in $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child’s\", \"your\") records via email, fax, or upload; working with Project Singular study staff to help you access $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child’s\", \"your\") records); or having our study staff contact $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child's\", \"your\") designated physician(s) and hospital(s)/institutions(s) to obtain the records).",
```
Comment (MocanaAtBroad) [inline pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf]: This was an error. Hopefully straightened out now.

File: pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf [hunk 3]
```diff
@@ -1658,9 +1658,9 @@
...
+        "upload": "I will upload the records myself via the Medical Record Upload button on $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"my child's\", \"my\") Project Singular Dashboard. This will include $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"my child's\", \"my\") most recent cardiology clinic note, echo report, catheterization report, and operation notes."
+        "fax": "I will fax or email them to the team at Boston Children’s Hospital, fax to (617) 730-6201 or email to <a href=\"mailto:project_singular@childrens.harvard.edu\" class=\"Link\">project_singular@childrens.harvard.edu</a>. This will include $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"my child's\", \"my\") most recent cardiology clinic note, echo report, catheterization report, and operation notes."
       },
```
Comment (MocanaAtBroad) [inline pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf:1664]: The email we had was wrong. Referred back to document with ticket. The correct address here is this childrens.harvard.edu address.

File: pepper-apis/studybuilder-cli/studies/singular/patches/medical-record-release_v2.conf [hunk 4]
```diff
@@ -1127,7 +1206,32 @@
...
+            "templateType": "HTML",
+            "templateText": """
+              <p class="paragraph">$ddp.date("MM / dd / yyyy")</p>
+            """,
+            "variables": [],
+          },
+        }
+
+        // Signature of parent or legally authorized representative: MRR_PARENT_SIGNATURE_NEW
         {
```
Comment (MocanaAtBroad) [inline pepper-apis/studybuilder-cli/studies/singular/patches/medical-record-release_v2.conf:1235]: Looks like a lot of changes, but essentially I added a new field (MRR_SIGNATURE) and did some reordering. Tried to get Date to show up behind name.

**Comment by Trimax** [comment]
> More fixes to medical release v2 There are a couple of problems remaining with this activity that I need some help with. One is that ticket requests we do the age checking to ensure AOM rules and workflow are not violated. @Trimax I would welcome your assistance here. I expect the checks you have been adding to consent/assent would be the same here. The other problem that I just noticed is that we have required questions in the first page; we need to make sure that either there is a picklist item from one question chosen (MRR_RECORD_SEND_OPTIONS) or the single picklist item (MRR_STAFF_GETS_RECORDS_DIRECTLY) from another question is chosen. I am not sure how we have done this in the past. @ssettipalli . I am hoping you can provide some guidance on this item.

Will be glad to help. Could you please give some details on that?

File: pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf [hunk 5]
```diff
@@ -1623,19 +1623,19 @@
...
     }
     "s3": {
       "p1": "$ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"By completing this information, you are agreeing to allow us to contact these physician(s) and hospital(s)/institutions(s) to obtain your child’s records. If you decide to collect and upload your child’s medical records to Project Singular, you are giving permission to Project Singular study staff to obtain your records.\", \"C. By completing this information, you are agreeing to allow us to contact these physician(s) and hospital(s)/institutions(s) to obtain your records. If you decide to collect and send your medical records to Project Singular study staff you are giving permission to Project Singular study staff to obtain your records.\")",
```
Comment (bskinner) [inline pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf]: I think there's a pre-existing typo here
```suggestion
    "s3": {
      "p1": "$ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"By completing this information, you are agreeing to allow us to contact these physician(s) and hospital(s)/institution(s) to obtain your child’s records. If you decide to collect and upload your child’s medical records to Project Singular, you are giving permission to Project Singular study staff to obtain your records.\", \"C. By completing this information, you are agreeing to allow us to contact these physician(s) and hospital(s)/institution(s) to obtain your records. If you decide to collect and send your medical records to Project Singular study staff you are giving permission to Project Singular study staff to obtain your records.\")",
```

File: pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf (hunk 2 above)
Comment (bskinner) [inline pepper-apis/studybuilder-cli/studies/singular/i18n/en.conf]: Unnecessary `s` after `institutions`
```suggestion
      "titleText": "By completing this information, you are agreeing to allow us to receive $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child’s\", \"your\") medical records through the option you selected above (sending in $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child’s\", \"your\") records via email, fax, or upload; working with Project Singular study staff to help you access $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child’s\", \"your\") records); or having our study staff contact $ddp.checkAnswer(\"MRR_WHO_ENROLLING_COPY\",\"CHILD\", \"your child's\", \"your\") designated physician(s) and hospital(s)/institution(s) to obtain the records).",
```

**Review by bskinner — DISMISSED** [review]
Changes looks good- did have a couple of suggestions for potential typos. As for the question about the exclusive-or picklist questions, what comes to mind using our current architecture is to use a PEX validation on both questions to check if either question has `isAnswered()` set to `true`:
```
(user.studies["singular"].forms["MEDICAL_RECORD_RELEASE"].questions["MRR_RECORD_SEND_OPTIONS"].isAnswered() &&
    !user.studies["singular"].forms["MEDICAL_RECORD_RELEASE"].questions["MRR_STAFF_GETS_RECORDS_DIRECTLY"].isAnswered()) ||
(!user.studies["singular"].forms["MEDICAL_RECORD_RELEASE"].questions["MRR_RECORD_SEND_OPTIONS"].isAnswered() &&
    user.studies["singular"].forms["MEDICAL_RECORD_RELEASE"].questions["MRR_STAFF_GETS_RECORDS_DIRECTLY"].isAnswered())
```

Not sure if we have any other lego pieces we can use here.

**Comment by ssettipalli** [comment]
We can do as Blake suggested 
It  can also be some activity validations like we have in singular/study-activities.conf 
I believe dmitri used these sort of validations in his last PR where he checked AGE/DOB in consent.
check if questions["QSTABLEID"].isAnswered() and check for options.

**Review by Trimax — APPROVED** [review]

**Comment by Trimax** [comment]
General question. How can I reach this form in the DEV environment? What path should I go on?

**Pull request merged**